### PR TITLE
feat: $pull operator support

### PR DESCRIPTION
**Pull request by fredmaggiowski**
## Pull Request Type

Introduce support for $pull operator

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Description

Address #64 

## PR Checklist

- [X] The commit message follows our guidelines included in the [CONTRIBUTING.md](../CONTRIBUTING.md#how-to-submit-a-pr)
- [X] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: tests/httpInterface.patchById.test.js [hunk 1]
```diff
@@ -379,6 +379,40 @@ tap.test('HTTP PATCH /<id>', async t => {
...
+        ...HTTP_DOC,
+        attachments: [{
+          name: 'another-note',
+          other: 'stuff',
+        }],
+        updaterId: newUpdaterId,
+      },
+    },
+    {
+      name: 'support $pull no duplicates',
```
Comment (danibix95) [inline tests/httpInterface.patchById.test.js:402]: This test is not clear to me. Could you clarify the description?

**Review by danibix95 — APPROVED** [review]

**Comment by danibix95** [comment]
LGTM. Just a small question regarding a test

File: tests/expectedSchemas/booksPatchBulkSchema.js [hunk 2]
```diff
@@ -905,14 +905,49 @@ module.exports = {
...
+                    },
+                    'anotherObject': {
+                      'type': 'object',
+                      'nullable': true,
+                    },
+                  },
+                  'additionalProperties': true,
                 },
-                'attachments\\.\\d+\\.neastedArr\\.\\$\\.replace$': {
+                'metadata.somethingArrayOfNumbers.$.replace': {
```
Comment (fredmaggiowski) [inline tests/expectedSchemas/booksPatchBulkSchema.js:943]: 🤔 this seems like an error

File: tests/httpInterface.patchBulk.test.js [hunk 3]
```diff
@@ -632,6 +632,129 @@ tap.test('HTTP PATCH /bulk', async t => {
...
+          },
+        ],
+      })
+
+      t.test('should update one document', t => {
+        t.equal(JSON.parse(response.payload), 1)
+        t.end()
+      })
+
+      t.test('should update the document without duplicates', async t => {
```
Comment (danibix95) [inline tests/httpInterface.patchBulk.test.js:676]:
```suggestion
      t.test('should update the document removing values from arrays', async t => {
```

File: tests/httpInterface.patchBulk.test.js [hunk 4]
```diff
@@ -632,6 +632,129 @@ tap.test('HTTP PATCH /bulk', async t => {
...
```
Comment (danibix95) [inline tests/httpInterface.patchBulk.test.js:737]:
```suggestion
      t.test('should update the document removing values from arrays', async t => {
```

**Pull request merged**